docs/best-practices.md

```diff
@@ -0,0 +1,799 @@
+---
+slug: /best-practices
+---
+
+# Best practices
+
+## Main structure and persistent collections
+
+The main contract structure is marked with `#[near_bindgen]`. It has to be serializable and deserializable with [Borsh](https://borsh.io).
+
+```rust
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct Contract {
+    pub data: String,
+    pub owner_id: AccountId,
+    pub value: u128,
+}
+```
+
+Every time an external method is called, the entire structure has to be deserialized.
+The serialized contract data is stored in [persistent storage] under the key `STATE`.
+
+Change methods ([see below](#view-vs-change-method)) serialize the main contract structure at the end and store the new value into storage.
+
+[Persistent collections](https://docs.rs/near-sdk/latest/near_sdk/collections) help store extra data in persistent storage outside of the main structure.
+NEAR SDK provides the following collections:
+
+- `Vector` - An iterable implementation of vector.
+- `LookupMap` - An non-iterable implementation of a map.
+- `LookupSet` - An non-iterable implementation of a set.
+- `UnorderedMap` - An iterable implementation of a map.
+- `UnorderedSet` - An iterable implementation of a set.
+- `TreeMap` - An iterable sorted map based on AVL-tree
+- `LazyOption` - An `Option` for a single value.
+
+Every instance of a persistent collection requires a unique storage prefix.
+The prefix is used to generate internal keys to store data in persistent storage.
+These internal keys need to be unique to avoid collisions (including collisions with key `STATE`).
```

Comment:
This is also already documented unless you want to add points to the `Collections` section that aren't there.

Reply:
Links and cleanup up: https://github.com/near/sdk-docs/pull/37/commits/f36a8eac128b98bf87a1b9e587b78214582995c3